samples/java-spring-cinema-booking-choreography/src/main/java/com/example/cinema/model/Show.java

```diff
@@ -0,0 +1,220 @@
+            case ShowEvent.SeatReserved seatReserved -> applyReserved(seatReserved);
+            case ShowEvent.SeatReservationPaid seatReservationPaid ->
+                    applyReservationPaid(seatReservationPaid);
+            case ShowEvent.SeatReservationCancelled seatReservationCancelled ->
+                    applyReservationCancelled(seatReservationCancelled);
+//            case ShowEvent.CancelledReservationConfirmed __ -> this;
+        };
+    }
+
+    public Show applyReserved(ShowEvent.SeatReserved seatReserved) {
+        Seat seat = getSeatOrThrow(seatReserved.seatNumber());
+        return new Show(id, title, seats.put(seat.number(), seat.reserved()),
+                pendingReservations.put(seatReserved.reservationId(), seatReserved.seatNumber()),
+                finishedReservations,seatReserved.availableSeatsCount());
+    }
+
+    public Show applyReservationPaid(ShowEvent.SeatReservationPaid seatReservationPaid) {
+        Seat seat = getSeatOrThrow(seatReservationPaid.seatNumber());
+        String reservationId = seatReservationPaid.reservationId();
+        FinishedReservation finishedReservation = new FinishedReservation(reservationId, seat.number()/*, CONFIRMED*/);
+        return new Show(id, title, seats.put(seat.number(), seat.paid()),
+                pendingReservations.remove(reservationId),
+                finishedReservations.put(reservationId, finishedReservation),availableSeats());
+
+    }
+
+    public Show applyReservationCancelled(ShowEvent.SeatReservationCancelled seatReservationCancelled) {
+        Seat seat = getSeatOrThrow(seatReservationCancelled.seatNumber());
+        String reservationId = seatReservationCancelled.reservationId();
+        FinishedReservation finishedReservation = new FinishedReservation(reservationId, seat.number()/*, CANCELLED*/);
+        return new Show(id, title, seats.put(seat.number(), seat.available()),
+                pendingReservations.remove(reservationId),
+                finishedReservations.put(reservationId, finishedReservation),seatReservationCancelled.availableSeatsCount());
+    }
+
+
+    private Seat getSeatOrThrow(int seatNumber) {
+        return seats.get(seatNumber).getOrElseThrow(() -> new IllegalStateException("Seat not found %s".formatted(seatNumber)));
+    }
+
+    public Option<Seat> getSeat(int seatNumber) {
+        return seats.get(seatNumber);
+    }
+
+    public enum SeatReservationStatus {
+      STARTED, SEAT_RESERVED, WALLET_CHARGE_REJECTED, WALLET_CHARGED, COMPLETED, SEAT_RESERVATION_FAILED, WALLET_REFUNDED, SEAT_RESERVATION_REFUNDED
+    }
+
+    public enum SeatStatus {
+      AVAILABLE, RESERVED, PAID
+    }
+
+    public record FinishedReservation(String reservationId, int seatNumber) { }
+
+    public record Reservation(String reservationId, String showId, String walletId, BigDecimal price) {}
+
+    public record Seat(int number, SeatStatus status, BigDecimal price) {
+      @JsonIgnore
```

Comment:
Is this really needed? It's kind of strange.

A record can't have extra fields. All fields should be in the constructor. So jackson should be able to understand that `isAvailable` is a method and therefore doesn't need to be added to the payload.

Reply:
Yes, it is required, we can probably disable this "feature" globally in the jackson configuration and use fields instead of accessor methods for serialization.

Reply:
If that's the case, better to keep it. Other users may find it useful. 

Feels like Jackson didn't catch up yet with records. This should not be applicable to records because only constructor fields can be used when deserializing so it doesn't make sense to serialize a method.